grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/Utility.java

```diff
@@ -142,4 +142,11 @@ public static ConceptId getResourceEdgeId(GraknTx graph, ConceptId conceptId1, C
         }
         return null;
     }
+
+    /**
+     * Get the id of given vertex.
+     */
+    static String getVertexId(Vertex vertex) {
+        return vertex.value(Schema.VertexProperty.ID.name());
+    }
```

Comment:
Can we wrap this in `ConceptId#of` immediately? If I remember right there were serialisation issues, but I still think it would be better to unwrap it before putting it in memory like `memory.put(KEY, cid.getValue());`

Reply:
I could, but then we need to do this wrapping and unwrapping a couple of times for each of the vertex in each iteration. Or do you meaning doing this only when adding to memory?